tools/nite/main.cpp

```diff
@@ -0,0 +1,386 @@
+			const_cast<std::remove_const_t<decltype(m_surface)>&>(m_surface) = nbl::video::CSimpleResizeSurface<nbl::video::CDefaultSwapchainFramebuffers>::create(std::move(surface));
+		}
+
+		if (m_surface)
+			return { {m_surface->getSurface()/*,EQF_NONE*/} };
+
+		return {};
+	}
+
+	inline bool onAppInitialized(smart_refctd_ptr<ISystem>&& system) override
+	{
+		m_inputSystem = make_smart_refctd_ptr<InputSystem>(logger_opt_smart_ptr(smart_refctd_ptr(m_logger)));
+
+		if (!device_base_t::onAppInitialized(smart_refctd_ptr(system)))
+			return false;
+
+		m_semaphore = m_device->createSemaphore(m_submitIx);
+		if (!m_semaphore)
+			return logFail("Failed to Create a Semaphore!");
+
+		ISwapchain::SCreationParams swapchainParams = { .surface = m_surface->getSurface() };
+		if (!swapchainParams.deduceFormat(m_physicalDevice))
+			return logFail("Could not choose a Surface Format for the Swapchain!");
+
+		const static IGPURenderpass::SCreationParams::SSubpassDependency dependencies[] =
+		{
+			{
+				.srcSubpass = IGPURenderpass::SCreationParams::SSubpassDependency::External,
+				.dstSubpass = 0,
+				.memoryBarrier =
+				{
+					.srcStageMask = asset::PIPELINE_STAGE_FLAGS::COPY_BIT,
+					.srcAccessMask = asset::ACCESS_FLAGS::TRANSFER_WRITE_BIT,
+					.dstStageMask = asset::PIPELINE_STAGE_FLAGS::COLOR_ATTACHMENT_OUTPUT_BIT,
+					.dstAccessMask = asset::ACCESS_FLAGS::COLOR_ATTACHMENT_WRITE_BIT
+				}
+			},
+			{
+				.srcSubpass = 0,
+				.dstSubpass = IGPURenderpass::SCreationParams::SSubpassDependency::External,
+				.memoryBarrier =
+				{
+					.srcStageMask = asset::PIPELINE_STAGE_FLAGS::COLOR_ATTACHMENT_OUTPUT_BIT,
+					.srcAccessMask = asset::ACCESS_FLAGS::COLOR_ATTACHMENT_WRITE_BIT
+				}
+			},
+			IGPURenderpass::SCreationParams::DependenciesEnd
+		};
+
+		auto scResources = std::make_unique<CDefaultSwapchainFramebuffers>(m_device.get(), swapchainParams.surfaceFormat.format, dependencies);
+		auto* renderpass = scResources->getRenderpass();
+
+		if (!renderpass)
+			return logFail("Failed to create Renderpass!");
+
+		auto gQueue = getGraphicsQueue();
+		if (!m_surface || !m_surface->init(gQueue, std::move(scResources), swapchainParams.sharedParams))
+			return logFail("Could not create Window & Surface or initialize the Surface!");
+
+		m_maxFramesInFlight = m_surface->getMaxFramesInFlight();
+		if (FRAMES_IN_FLIGHT < m_maxFramesInFlight)
+		{
+			m_logger->log("Lowering frames in flight!", ILogger::ELL_WARNING);
+			m_maxFramesInFlight = FRAMES_IN_FLIGHT;
+		}
+
+		m_cmdPool = m_device->createCommandPool(gQueue->getFamilyIndex(), IGPUCommandPool::CREATE_FLAGS::RESET_COMMAND_BUFFER_BIT);
+
+		for (auto i = 0u; i < m_maxFramesInFlight; i++)
+		{
+			if (!m_cmdPool)
+				return logFail("Couldn't create Command Pool!");
+			if (!m_cmdPool->createCommandBuffers(IGPUCommandPool::BUFFER_LEVEL::PRIMARY, { m_cmdBufs.data() + i, 1 }))
+				return logFail("Couldn't create Command Buffer!");
+		}
```

Comment:
you should probably create all commandbuffers at once

Reply:
we agreed to fix/update/adjust the nite in the next PR